moveit_experimental/moveit_jog_arm/src/cpp_interface_example/cpp_interface_example.cpp

```diff
@@ -49,7 +49,7 @@ int main(int argc, char** argv)
 
   // Run the jogging C++ interface in a new thread to ensure a constant outgoing message rate.
   moveit_jog_arm::JogCppApi jog_interface;
-  std::thread jogging_thread(&moveit_jog_arm::JogCppApi::mainLoop, &jog_interface);
+  std::thread jogging_thread([&]() { jog_interface.startMainLoop(); });
```

Comment:
This syntax is weird to me. What's going on here?

Is this similar to `this`?

Reply:
This is a [lambda](https://en.cppreference.com/w/cpp/language/lambda).